### PR TITLE
Yaml validation

### PR DESCRIPTION
**Pull request by KirbyBroderick**
This might break a lot of things, but the best way to test is to have people actually run things. Fortunately, the validation package is really intuitive and powerful. Hopefully this makes it a lot more obvious when people should change their configs.

### CODE REVIEW

File: catlas/filter_utils.py [hunk 1]
```diff
@@ -12,6 +12,13 @@
     PourbaixDiagram,
     PourbaixEntry,
 )
+import cerberus
+
+
+def validate_config(yaml: dict) -> None:
```
Comment (brookwander) [inline catlas/filter_utils.py:18]: Am I missing something here?

File: catlas/validation.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
+from pymatgen.core import Element
+from cerberus import Validator
+from pathlib import Path
+import yaml
+import os
+
```
Comment (brookwander) [inline catlas/validation.py:6]: probably rename this to be more intuitive. Validation is very broad. Probably `input_yaml_validation.py`?

File: configs/automated_screens/intermetallics_CO2RR.yml [hunk 3]
```diff
@@ -1,16 +1,12 @@
-dask:
-  partitions: -1 # Use automatic guess for number of partitions
-
 memory_cache_location: '/home/jovyan/shared-scratch/catlas/cache_github'
```
Comment (brookwander) [inline configs/automated_screens/intermetallics_CO2RR.yml:1]: I am pretty sure this file is practically deprecated. Maybe we should get rid of it?

File: configs/dask_cluster/dev_kube_cluster/worker-cpu.tmpl [hunk 4]
```diff
@@ -8,9 +17,9 @@ spec:
     name: dask-worker
     env: 
     - name: OMP_NUM_THREADS
-      value: 4
+      value: 1
     - name: MP_API_KEY
-      value: "your_api_key"
+      value: "iHay4qJxqBUVpmbX20XJ43gQfXI77JhF"
```
Comment (brookwander) [inline configs/dask_cluster/dev_kube_cluster/worker-cpu.tmpl:22]: this should be a secret

File: configs/dask_cluster/dev_kube_cluster/worker-gpu.tmpl [hunk 5]
```diff
@@ -14,7 +14,7 @@ spec:
     - name: MALLOC_TRIM_THRESHOLD_
       value: 0
     - name: MP_API_KEY
-      value: "your_api_key"
+      value: "iHay4qJxqBUVpmbX20XJ43gQfXI77JhF"
```
Comment (brookwander) [inline configs/dask_cluster/dev_kube_cluster/worker-gpu.tmpl:17]: again

File: configs/dask_cluster/local_cluster/local_cluster.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
 from dask.distributed import Client, LocalCluster
 
-cluster = LocalCluster(n_workers=2, threads_per_worker=1)
+cluster = LocalCluster(n_workers=4, threads_per_worker=1)
```
Comment (brookwander) [inline configs/dask_cluster/local_cluster/local_cluster.py:3]: why?

File: catlas/validation.py [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+config_schema = {
+    "memory_cache_location": {"type": "string", "check_with": validate_folder_exists},
+    "input_options": {
+        "required": True,
+        "type": "dict",
+        "schema": {
+            "adsorbate_file": {
+                "required": False,
```
Comment (brookwander) [inline catlas/validation.py:58]: I think this is required? We dont support just slab enumeration now

File: catlas/validation.py [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+                "type": "list",
+                "allowed": valid_element_groups,
+            },
+            "filter_by_pourbaix_stability": {
+                "type": "dict",
+                "schema": {
+                    "lmdb_path": {
+                        "required": True,
+                        "type": "string",
+                        # "check_with": validate_folder_exists,
```
Comment (brookwander) [inline catlas/validation.py:109]: why is this commented out?

File: catlas/validation.py [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+                    },
+                    "conditions_list": {
+                        "required": True,
```
Comment (brookwander) [inline catlas/validation.py:112]: maybe I am misunderstanding but does this being required make the range option fail?

File: catlas/validation.py [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+                            "pH_upper",
+                            "pH_step",
+                            "V_lower",
+                            "V_upper",
+                            "V_step",
+                        ],
+                        "type": "list",
+                        "schema": {
+                            "type": "dict",
+                            "required": True,
```
Comment (brookwander) [inline catlas/validation.py:124]: same

**Review by brookwander — CHANGES_REQUESTED** [review]

File: catlas/filter_utils.py (hunk 1 above)
Comment (KirbyBroderick) [reply catlas/filter_utils.py:18]: What do you mean?

File: catlas/validation.py (hunk 9 above)
Comment (KirbyBroderick) [reply catlas/validation.py:112]: If two things exclude each other it acts like XOR

File: catlas/filter_utils.py (hunk 1 above)
Comment (brookwander) [reply catlas/filter_utils.py:18]: its a function that does nothing?

**Review by brookwander — APPROVED** [review]
This looks good the only exception being our unresolved comment thread.

File: catlas/filter_utils.py [hunk 11]
```diff
@@ -12,6 +12,7 @@
     PourbaixDiagram,
     PourbaixEntry,
 )
+import cerberus
```
Comment (zulissi) [inline catlas/filter_utils.py:15]: remove?

**Review by zulissi — APPROVED** [review]
LGTM, just one errant cerberus import?

**Pull request merged**